datasketch/lsh.py

```diff
@@ -113,22 +115,23 @@ def __init__(self, threshold=0.9, num_perm=128, weights=(0.5,0.5),
             self.prepickle = prepickle
         basename = _random_name(11)
         self.hashtables = [
-            unordered_storage(storage_config, name=basename + b'_bucket_' + bytes([i]))
+            unordered_storage(storage_config, name=basename + b'_bucket_' + bytes([i]), buffer_size=self._buffer_size)
             for i in range(self.b)]
         self.hashranges = [(i*self.r, (i+1)*self.r) for i in range(self.b)]
-        self.keys = ordered_storage(storage_config, name=basename + b'_keys')
+        self.keys = ordered_storage(storage_config, name=basename + b'_keys', buffer_size=self._buffer_size)
 
-    def insert(self, key, minhash):
+    def insert(self, key, minhash, check_duplication=True):
         '''
         Insert a unique key to the index, together
         with a MinHash (or weighted MinHash) of the set referenced by 
         the key.
 
         Args:
             key (hashable): The unique identifier of the set. 
-            minhash (datasketch.MinHash): The MinHash of the set. 
+            minhash (datasketch.MinHash): The MinHash of the set.
+            check_duplication (bool): check duplication.
```

Comment:
Shouldn't the key be always unique? If we allow non-unique keys then what would happen when there is a duplicate?

Reply:
Of couse yes. But we can manage it in the module or give client to manage it, to avoid seeking in storage.

Reply:
So if there is an duplication, the client does not check, and `check_duplication=False`, what would happen? If there is going to be an exception, then we don't really need to have this parameter. Also, more detailed description for this argument would be great.

Reply:
I exteded docs describing usage of this argument